tree/ntuple/CMakeLists.txt

```diff
@@ -89,6 +94,16 @@ if(daos OR daos_mock)
   endif()
 endif()
 
+# Enable RNTuple support for S3 backend through Davix
+if(davix OR builtin_davix)
+  set(ROOTNTuple_EXTRA_HEADERS ${ROOTNTuple_EXTRA_HEADERS} ROOT/RPageStorageS3.hxx)
+  target_sources(ROOTNTuple PRIVATE v7/src/RPageStorageS3.cxx)
+  target_compile_definitions(ROOTNTuple PRIVATE R__ENABLE_DAVIX)
+
+  target_include_directories(ROOTNTuple PRIVATE ${DAVIX_INCLUDE_DIR})
+  target_link_libraries(ROOTNTuple PRIVATE ${DAVIX_LIBRARY})
+endif()
+
```

Comment:
Now that we can foresee the development of other backends in the future, we might want to have a better way of specifying which backends get built, e.g. `-Drntuple_opt_backends=daos,s3` (the file backend should probably be always built).

Perhaps we could discuss the advantages at some point.